### PR TITLE
upgrade walletConnect ---> V2 

### LINKED ISSUE #177: Migrate Wallet Connect to v2
First reported in [Discord](https://discord.com/channels/690584551239581708/1136769849427771393).

Depends on #175

### PR DESCRIPTION
**Pull request by yahiamarzouk**
## Dework Task

-this fixes the walletConnect issue and now wallet connect V2 will be working
-when https://github.com/nation3/app/pull/178 is merged this should be merged with it.

https://app.dework.xyz/nation3/open-bounties-95838?taskId=3c1e59f0-92f2-4157-97c3-43f4918a4c43

## Related GitHub Issue

Fixes #177 177

## Screenshots (if appropriate):
Screenshot 2023-08-09 at 9 30 52 PM

## How Has This Been Tested?

his has been tested using trust wallet and meta mask wallet on an Iphone

https://github.com/nation3/app/assets/94828675/c036035a-caaf-405e-926f-ac05f8bfba2d

### CODE REVIEW

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
The Wallet Connect dialog is incorrectly showing _under_ the Nation3 dialog.  It should be the reverse.

>

**Comment by yahiamarzouk** [comment]
> The Wallet Connect dialog is incorrectly showing _under_ the Nation3 dialog. It should be the reverse.
> 
> Maybe a z-index problem.
> 
> > 


https://github.com/nation3/app/assets/94828675/9cb09cdb-25a6-4c7c-b046-1bf0d8e717b5

Done (Y)

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "@walletconnect/modal": "^2.6.1"
```
Comment (aahna-ashina) [inline package.json:3]: This code should be in the `/ui` folder.

**Review by aahna-ashina — CHANGES_REQUESTED** [review]

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
The `yarn.lock` file should be in the `ui` folder.

**Review by aahna-ashina — APPROVED** [review]

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**